server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v64/PopulateOrganizationMembersTableTest.java

```diff
@@ -0,0 +1,308 @@
+    setupDefaultOrganization();
+    underTest.execute();
+
+    underTest.execute();
+  }
+
+  @Test
+  public void migrate_active_users_to_default_organization() throws Exception {
+    setupDefaultOrganization();
+    int user1Id = insertUser(USER1_LOGIN, true);
+    int user2Id = insertUser(USER2_LOGIN, true);
+    int user3Id = insertUser(USER3_LOGIN, false);
+
+    underTest.execute();
+
+    verifyUserMembership(user1Id, DEFAULT_ORGANIZATION_UUID);
+    verifyUserMembership(user2Id, DEFAULT_ORGANIZATION_UUID);
+    verifyUserMembership(user3Id);
+  }
+
+  @Test
+  public void migrate_user_having_direct_global_permissions() throws Exception {
+    setupDefaultOrganization();
+    insertOrganization(ORG1_UUID);
+    insertOrganization(ORG2_UUID);
+    int userId = insertUser(USER1_LOGIN);
+    insertUserRole(userId, PERMISSION_PROVISIONING, ORG1_UUID, null);
+    insertUserRole(userId, PERMISSION_ADMIN, ORG1_UUID, null);
+    insertUserRole(userId, PERMISSION_ADMIN, ORG2_UUID, null);
+
+    underTest.execute();
+
+    verifyUserMembership(userId, ORG1_UUID, ORG2_UUID, DEFAULT_ORGANIZATION_UUID);
+  }
+
+  @Test
+  public void migrate_user_having_direct_project_permissions() throws Exception {
+    setupDefaultOrganization();
+    insertOrganization(ORG1_UUID);
+    insertOrganization(ORG2_UUID);
+    int userId = insertUser(USER1_LOGIN);
+    insertUserRole(userId, PERMISSION_BROWSE, ORG1_UUID, 1);
+    insertUserRole(userId, PERMISSION_CODEVIEWER, ORG1_UUID, 1);
+    insertUserRole(userId, PERMISSION_ADMIN, ORG2_UUID, 2);
+
+    underTest.execute();
+
+    verifyUserMembership(userId, ORG1_UUID, ORG2_UUID, DEFAULT_ORGANIZATION_UUID);
+  }
+
+  @Test
+  public void migrate_user_having_global_permissions_from_group() throws Exception {
+    setupDefaultOrganization();
+    insertOrganization(ORG1_UUID);
+    insertOrganization(ORG2_UUID);
+    int userId = insertUser(USER1_LOGIN);
+    int group1Id = insertNewGroup(ORG1_UUID);
+    int group2Id = insertNewGroup(ORG2_UUID);
+    insertUserGroup(userId, group1Id);
+    insertUserGroup(userId, group2Id);
+
+    underTest.execute();
+
+    verifyUserMembership(userId, ORG1_UUID, ORG2_UUID, DEFAULT_ORGANIZATION_UUID);
+  }
+
+  @Test
+  public void user_without_any_permission_should_be_member_of_default_organization() throws Exception {
+    setupDefaultOrganization();
+    int userId = insertUser(USER1_LOGIN);
+
+    underTest.execute();
+
+    verifyUserMembership(userId, DEFAULT_ORGANIZATION_UUID, DEFAULT_ORGANIZATION_UUID);
+  }
+
+  @Test
+  public void migrate_users_having_any_kind_of_permission() throws Exception {
+    setupDefaultOrganization();
+    insertOrganization(ORG1_UUID);
+    insertOrganization(ORG2_UUID);
+    int user1 = insertUser(USER1_LOGIN);
+    int user2 = insertUser(USER2_LOGIN);
+    int user3 = insertUser(USER3_LOGIN);
+    int groupId = insertNewGroup(ORG1_UUID);
+    insertUserGroup(user2, groupId);
```

Comment:
No need to add permission on the group => At long as a user belongs to a group of an orga, he becomes member of this orga, even without any permission on it